source/common/event/guarddog_impl.cc

```diff
@@ -0,0 +1,100 @@
+#include "common/common/assert.h"
+#include "common/event/guarddog_impl.h"
+#include "common/event/watchdog_impl.h"
+
+namespace Event {
+
+GuardDogImpl::GuardDogImpl(Stats::Store& stats_store, const Server::Configuration::Main& config,
+                           SystemTimeSource& tsource)
+    : time_source_(tsource), miss_timeout_(config.wdMissTimeout()),
+      megamiss_timeout_(config.wdMegaMissTimeout()), kill_timeout_(config.wdKillTimeout()),
+      multi_kill_timeout_(config.wdMultiKillTimeout()),
+      loop_interval_(
+          std::min({miss_timeout_, megamiss_timeout_,
+                    kill_enabled() ? kill_timeout_ : std::min(miss_timeout_, megamiss_timeout_),
```

Comment:
I didn't think that was possible since loop_interval_ is const? But, come to think of it, I could just write it as a lambda that executes and returns the value used in the initializer list, and the lambda could contain a somewhat simpler and easier to read version of this min mess.